src/middlewares/auth.js

```diff
@@ -0,0 +1,81 @@
+import { expressjwt } from 'express-jwt';
+import prisma from '../lib/prismaClient.js';
+import { AuthorizeError, BadRequestError } from '../lib/error.js';
+import jwt from 'jsonwebtoken';
+
+// 리퀘스트 토큰 검증 미들웨어
+const verifyRefreshToken = expressjwt({
+  secret: process.env.JWT_SECRET,
+  algorithms: ['HS256'],
+  getToken: (req) => req.cookies.refreshToken,
+});
+
+// 엑세스 토큰 검증 미들웨어
+const verifyAccessToken = expressjwt({
+  secret: process.env.JWT_SECRET,
+  algorithms: ['HS256'],
+  getToken: (req) => req.cookies.accessToken,
+});
+
+// 유저 인가 미들웨어
+async function authorizeUser(req, res, next) {
+  const { userId } = req.auth;
+  const user = await prisma.user.findUnique({ where: { id: userId } });
+  if (!user) next(new BadRequestError());
+  req.user = user;
+  next();
+}
+
+// 토큰 유무 확인하는 전역 미들웨어
+function optionalAuth(req, res, next) {
+  const token = req.cookies.accessToken;
+  if (!token) return next();
+  jwt.verify(token, process.env.JWT_SECRET, (err, decoded) => {
+    // 검증 실패시 에러, 성공시 payload 반환
+    if (!err) req.auth = decoded;
+    next();
```

Comment:
💊 구현이 잘못 된 것 같아요!
 if 를 inline으로 묶다보니 한줄만 실행될 것 같습니다 ㅎㅎ

```suggestion
    // 검증 실패시 에러, 성공시 payload 반환
    if (err) {
      next(new AuthorizeError());
      return;
    }
    req.auth = decoded;
    next();
```

Reply:
넵 감사합니다.  수정했습니다.